stocks2.lua

```diff
@@ -0,0 +1,544 @@
+-- a v50 script remake of the older stocks plugin
+-- chawkzero, Aug 2023
+
+local gui = require("gui")
+local widgets = require("gui.widgets")
+local utils = require("utils")
+
+local item_desc_width = 38
+local item_table = {}
+
+local current_search_query = ""
+local filter_states = {
+    in_job = true,
+    rotten = true,
+    owned = true,
+    forbidden = true,
+    dump = true,
+    on_fire = true,
+    melt = true,
+    in_inv = true,
+    caged = true,
+    trade = true,
+    no_flags = true,
+}
+
+local min_qual = 0
+local max_qual = 6
+local min_wear = 0
+
+-- 0 = listed, 1 = selected
+local apply_to_mode = 0
+local apply_to_labels = {"Listed", "Selected"}
+
+local stocks_window_width = 106
+local stocks_window_height = 50
+
+StocksWindow = defclass(StocksWindow, widgets.Window)
+StocksWindow.ATTRS({
+    frame = {w = stocks_window_width, h = stocks_window_height},
+    frame_title = "Stocks",
+    autoarrange_subviews = false,
+    autoarrange_gap = 0,
+})
+
+function StocksWindow:init()
+    self:addviews({
+        widgets.Panel({
+            view_id = "left_panel",
+            frame = {t = 0, b = 0, l = 0, r = stocks_window_width - 76},
+            frame_style = gui.INTERIOR_FRAME,
+            subviews = {
+                widgets.EditField({
+                    view_id = "search_box",
+                    frame = {t = 1, l = 0},
+                    label_text = "Search: ",
+                    on_char = function(ch) return ch:match("[%l%s]") end,
+                    on_change = function(new_text, old_text)
+                        current_search_query = new_text
+                        self:filter_list(current_search_query)
+                    end,
+                }),
+                widgets.Label({
+                    text = {{text = "Items: ", pen = COLOR_LIGHTGREEN}},
+                    frame = {t = 3, l = 0},
+                    view_id = "item_count_label"
+                }),
+                widgets.List({
```

Comment:
This should be a FilteredList instead of just a List so you don't have to manage searching yourself.